aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -272,7 +272,24 @@ std::vector<Tensor> where(const Tensor& condition) {
 }
 
 Tensor _s_where(const Tensor& condition, const Tensor& self, const Tensor& other) {
-  TORCH_CHECK(self.dtype() == other.dtype(), "expected scalar type ", self.dtype(), " but found ", other.dtype());
```

Comment:
The current error message can be improved by giving the name of the function, so an example string would be "torch.where expected scalar type ...", but I don't think we want to edit this check or the error message beyond that. Our error messages are all a little bit inconsistent, unfortunately, but they generally follow the pattern of: function name + expected ... + but got ... 

Especially since this function is probably being called from Python we don't want to include C++ code examples, and examples would be inconsistent with other error messages like this, anyway.

Reply:
I don't think the current error message can be improved in the way you suggest. `_s_where` is somewhere deep in the stack whereas the problem happens at a much higher level.

It's possible this issue can only arise from C++ code? I'm not sure. Please see D25915551 for context.

Reply:
I think you'll have to track down where this problematic scalar is being generated, then. This error message doesn't address the issue and would be inconsistent with our other error messages.

Maybe, however, there's a systemic way to improve the debugging information provided when we throw an error? I wouldn't want to change just this error message.